dipy/denoise/cnn_1denoiser/cnn_1denoiser.py

```diff
@@ -0,0 +1,317 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv1D, Activation
+from tensorflow.keras.models import *
+from tensorflow.keras.layers import Input
+#from patchify import patchify, unpatchify
+from sklearn.model_selection import train_test_split
+from datetime import datetime
+now = datetime.utcnow().strftime("%Y%m%d%H%M%S")
+root_logdir = "tf_logs"
+logdir = "{}/run-{}/".format(root_logdir, now)
+
+
+
+
+
+
+'''
+
+we constructed a simple 1D-CNN model that has five layers, including two convolutional layers, each
+followed by a max-pooling layer, and a dense layer (Figure 1). The first convolutional layer was given an
+input of the ‘noisy’ high-noise image and consisted of 16 one-dimensional filtering kernels of size 16. The
+second convolutional layer consisted of 32 one-dimensional filtering kernels of size 8. The ReLu
+activation function was used in both convolutional layers. There was 1 max-pooling layer that had a kernel
+size of 2 with stride 2. In the dense layer, the extracted features of the high-noise image were mapped to
+the low-noise reference image.
+
+
+
+
+Contributed by :
+Hu Chenga,b Sophia Vinci-Boohera, Jian Wangc, Bradley Carona, Qiuting Wend, Sharlene Newmane,Franco Pestillif
+
+Ref:Denoising diffusion weighted imaging data using convolutional neural networks( https://www.biorxiv.org/content/10.1101/2022.01.17.476708v1.abstract )
+Ref Code : https://github.com/huchengMRI/DWI-SoS-denoising
```

Comment:
Please refer to the published version: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0274396